src/functionaltest/java/com/ericsson/ei/frontend/TestRulesFunctionality.java

```diff
@@ -0,0 +1,89 @@
+package com.ericsson.ei.frontend;
+
+import static org.junit.Assert.assertEquals;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import com.ericsson.ei.config.SeleniumConfig;
+import com.ericsson.ei.frontend.pageobjects.IndexPage;
+import com.ericsson.ei.frontend.pageobjects.TestRulesPage;
+
+public class TestRulesFunctionality extends SeleniumBaseClass {
+
+    @Test
+    public void testJourneyToFindAggregatedObjectButton() throws Exception {
+        // Load index page and wait for it to load
+        IndexPage indexPageObject = new IndexPage(mockedHttpClient, driver, baseUrl);
+        indexPageObject.loadPage();
+        TimeUnit.SECONDS.sleep(3);
+
+        // Verify that we can navigate to test rules page
+        TestRulesPage testRulesPage = indexPageObject.clickTestRulesPage();
+        new WebDriverWait(driver, 10).until((webdriver) -> testRulesPage.presenceOfTestRulesHeader());
+
+        // Verify that "download rules template" button works
+        String mockedResponse = this.getJSONStringFromFile(
+                "src/functionaltest/resources/responses/RulesTemplateObject.json");
+        TimeUnit.SECONDS.sleep(1);
```

Comment:
Is this sleep really needed? It should not reach this point unless mockResponse is ready.

Reply:
It's not mockedresponse we are waiting for, it's the "download rules template" thats has to be present on the website, otherwise the click fails. I believe i tried with a "wait until download rules template is present" before and thats the reason i had to make a sleep instead but i might be mistaken, i will try with that again if im mistaken.